diff --git a/includes/export.inc b/includes/export.inc
index 0b85c2e..e2e6013 100644
--- a/includes/export.inc
+++ b/includes/export.inc
@@ -817,6 +817,9 @@ function _ctools_export_unpack_object($schema, $data, $object = 'stdClass') {
   // Go through our schema and build correlations.
   foreach ($schema['fields'] as $field => $info) {
     if (isset($data->$field)) {
+      if (!empty($info['serialize']) && !is_serialized($data->$field)) {
+        $data->$field = serialize($data->$field);
+      }
       $object->$field = empty($info['serialize']) ? $data->$field : unserialize($data->$field);
     }
     else {
@@ -839,6 +842,32 @@ function _ctools_export_unpack_object($schema, $data, $object = 'stdClass') {
   return $object;
 }
 
+function _ctools_is_serialized($data){
+  // if it isn't a string, it isn't serialized
+  if ( !is_string( $data ) )
+    return false;
+  $data = trim( $data );
+  if ( 'N;' == $data )
+    return true;
+  if ( !preg_match( '/^([adObis]):/', $data, $badions ) )
+    return false;
+  switch ( $badions[1] ) {
+    case 'a' :
+    case 'O' :
+    case 's' :
+      if ( preg_match( "/^{$badions[1]}:[0-9]+:.*[;}]\$/s", $data ) )
+        return true;
+      break;
+    case 'b' :
+    case 'i' :
+    case 'd' :
+      if ( preg_match( "/^{$badions[1]}:[0-9.E-]+;\$/", $data ) )
+        return true;
+      break;
+  }
+  return false;
+}
+
 /**
  * Unpack data loaded from the database onto an object.
  *
