diff --git a/quicktabs.admin.inc b/quicktabs.admin.inc
index 6b108f9..e06dc7f 100644
--- a/quicktabs.admin.inc
+++ b/quicktabs.admin.inc
@@ -239,10 +239,17 @@ function _quicktabs_admin_main_form($form_state, &$qt) {
     '#weight' => -4,
   );
 
+  $form['do_not_log_no_content'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Do not log errors for empty quicktabs'),
+    '#default_value' => isset($qt->do_not_log_no_content) ? $qt->do_not_log_no_content : 0,
+    '#description' => t('When there are empty views in a quicktab, an error is thrown. This prevent that error from being sent to Watchdog.'),
+    '#weight' => -3,
+  );
   // Add a wrapper for the tabs and Add Another Tab button.
   $form['qt_wrapper'] = array(
     '#tree' => FALSE,
-    '#weight' => -3,
+    '#weight' => -2,
     '#prefix' => '<div class="clear-block" id="quicktabs-tabs-wrapper">',
     '#suffix' => '</div>',
   );
@@ -648,6 +655,7 @@ function _quicktabs_convert_form_to_quicktabs($form_state) {
   $qt->style = $form_state['values']['style'];
   $qt->tabs = $formvalues_tabs;
   $qt->options = isset($form_state['values']['options'][$renderer]) ? $form_state['values']['options'][$renderer] : array();
+  $qt->options['do_not_log_no_content'] = $form_state['values']['do_not_log_no_content'];
 
   if (isset($form_state['values']['machine_name'])) {
     $qt->machine_name = $form_state['values']['machine_name'];
diff --git a/quicktabs.classes.inc b/quicktabs.classes.inc
index 56c2569..ee16162 100644
--- a/quicktabs.classes.inc
+++ b/quicktabs.classes.inc
@@ -75,7 +75,9 @@ class QuickSet {
         $qs = new self($name, $contents, $settings);
       }
       catch (InvalidQuickSetException $e) {
-        watchdog('Quicktabs', $e->getMessage());
+        if (!(array_key_exists('do_not_log_no_content', $settings['options']) && $settings['options']['do_not_log_no_content'] && $e->getMessage() == InvalidQuickSetException::NO_CONTENT)) {
+          watchdog('Quicktabs', $e->getMessage());
+        }
         return NULL;
       }
       return new $class($qs);
@@ -118,7 +120,7 @@ class QuickSet {
       'hide_empty_tabs' => 0,
       'ajax' => 0,
       'default_tab' => 0,
-      'options' => array(),
+      'options' => array('do_not_log_no_content' => 0,),
     );
   }
   
@@ -181,7 +183,7 @@ class QuickSet {
    */
   protected function prepareContents() {
     if (!count($this->contents))  {
-      throw new InvalidQuickSetException('There are no contents to render.');
+      throw new InvalidQuickSetException(InvalidQuickSetException::NO_CONTENT);
     }
     if ($this->settings['hide_empty_tabs'] && !$this->settings['ajax']) {
       // Check if any tabs need to be hidden because of empty content.
@@ -203,7 +205,7 @@ class QuickSet {
         }
       }
       if (!$renderable_contents)  {
-        throw new InvalidQuickSetException('There are no contents to render.');
+        throw new InvalidQuickSetException(InvalidQuickSetException::NO_CONTENT);
       }
     }
     elseif ($this->settings['ajax']) {
@@ -512,5 +514,5 @@ class QuickPreRenderedContent implements QuickContentRenderable {
  * Create our own exception class.
  */
 class InvalidQuickSetException extends Exception {
-  
+  const NO_CONTENT = 'There are no contents to render.';
 }
\ No newline at end of file
