diff --git a/redirect.admin.inc b/redirect.admin.inc
index 176f105..7fc4194 100644
--- a/redirect.admin.inc
+++ b/redirect.admin.inc
@@ -469,7 +469,10 @@ function redirect_element_validate_redirect($element, &$form_state) {
   // Normalize the path.
   $value = drupal_get_normal_path($value, $form_state['values']['language']);
 
-  if (!valid_url($value) && !valid_url($value, TRUE) && $value != '<front>' && $value != '' && !file_exists($value)) {
+  // We use the encoded URL for most checks, except the check against <front>
+  $testvalue = drupal_encode_path($value);
+
+  if (!valid_url($testvalue) && !valid_url($testvalue, TRUE) && $value != '<front>' && $testvalue != '') {
     form_error($element, t('The redirect path %value is not valid.', array('%value' => $value)));
   }
 
